### ISSUE TITLE
"none" item, unknown cause; possibly smoking rack

### ISSUE THREAD

**Issue by Funguss** [CleverRaven/Cataclysm-DDA#38462]
As title. Find save below. There was an error when removing charcoal from the smoking rack, but I must admit I didn't pay attention.

[Bentonia.zip](https://github.com/CleverRaven/Cataclysm-DDA/files/4270611/Bentonia.zip)

# Steps To Reproduce

???

# Screenshots

![none](https://user-images.githubusercontent.com/57632331/75611893-6b8c8200-5b16-11ea-8e69-789147fed7a7.png)

# Versions and configuration

- OS: Windows
    - OS Version: MINGW/CYGWIN/MSYS2 on unknown Windows version
- Game Version: 0.D-12504-g9015618 [64-bit]
- Graphics Version: Tiles
- Mods loaded: [
    Dark Days Ahead [dda],
    Aftershock [aftershock],
    C.R.I.T Expansion Mod [crt_expansion],
    Magiclysm [magiclysm],
    Makeshift Items Mod [makeshift],
    More Survival Tools [More_Survival_Tools],
    Mythological Replicas [nw_pack],
    Hydroponics [hydroponics],
    Mutant NPCs [mutant_npcs],
    No Rail Stations [No_Rail_Stations],
    Folding Parts pack [deoxymod],
    Vehicle Additions Pack [blazemod],
    Tanks and Other Vehicles [Tanks],
    Roadheader and other mining vehicles [Heavy miners],
    No Fungal Monsters [No_Fungi],
    Beta National Guard Camp [national_guard_camp]
]

**Comment by jkraybill** [comment]
I got this same issue just once. I think there is a bug to do with loading charcoal into a smoking rack that already has charcoal in it. Will try to find the repro for this.

**Comment by Funguss** [comment]
Right, I did do that!

**Comment by jkraybill** [comment]
OK, I've got the base repro.

Build a smoking rack
(E)xamine it, (r)eload it with one charcoal
Insert 1 smokable thing into the smoker (I used strawberries)
(E)xamine it again, (r)eload with charcoal
When it asks you "The smoking rack contains 1 charcoal. Unload?" say Yes.
The error occurs (debug log attached):

![image](https://user-images.githubusercontent.com/1361701/75619734-d059e200-5bd3-11ea-86e2-7bcb60577c74.png)

After skipping the error, you now have a blank space and a "none" item in your inventory.

The game allows you to drop the "none", and even pick it up again.

[debug.log](https://github.com/CleverRaven/Cataclysm-DDA/files/4271358/debug.log)
